api/controller.py

```diff
@@ -732,7 +733,9 @@ def search(self, lane_identifier):
         language_header = flask.request.headers.get("Accept-Language")
         if language_header:
             languages = parse_header(language_header)
-            languages = map(lambda x: str(x), languages)
+            languages = map(str, languages)
+            languages = map(LanguageCodes.iso_639_2_for_locale, languages)
+            languages = [l for l in languages if l]
```

Comment:
Is there a better way to do the previous two lines? The `Accept-Language` header after mapping it to `str` was `['en-US', 'en; q=0.90']`, but running that through `iso_639_2_for_locale` turned the second value into `None` which raised a parsing error in a later method.

Reply:
parse_header returns a list of AcceptableType objects. You can access the language part as `AcceptableType.mime_type`. That's not a great name because parse_header was designed to parse the `Accept` header, but the format is the same, with the language code standing in for the MIME type.